google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupRemarketing.java

```diff
@@ -0,0 +1,451 @@
+      // Adds the ad group criterion.
+      MutateAdGroupCriteriaResponse response = adGroupCriterionServiceClient.mutateAdGroupCriteria(
+          Long.toString(customerId),
+          ImmutableList.of(operation));
+      // Gets and prints the results.
+      String adGroupCriterionResourceName = response.getResults(0).getResourceName();
+      System.out.printf("Successfully created ad group criterion with resource name '%s' "
+          + "targeting user list with resource name '%s' with ad group with ID %d.%n",
+          adGroupCriterionResourceName,
+          userList,
+          adGroupId);
+      return adGroupCriterionResourceName;
+    }
+  }
+
+  /**
+   * Updates the bid modifier on an ad group criterion.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroupCriterionResourceName the ad group criterion to update.
+   * @param bidModifier the bid modifier.
+   */
+  private void modifyAdGroupBids(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      String adGroupCriterionResourceName,
+      double bidModifier) {
+    // Creates the ad group criterion with a bid modifier of 1.25. You may alternatively set the
+    // bid for the ad group criterion directly.
+    AdGroupCriterion adGroupCriterion = AdGroupCriterion.newBuilder()
+        .setResourceName(adGroupCriterionResourceName)
+        .setBidModifier(DoubleValue.of(bidModifier))
+        .build();
+
+    // Creates the update operation.
+    AdGroupCriterionOperation operation = AdGroupCriterionOperation.newBuilder()
+        .setUpdate(adGroupCriterion)
+        .setUpdateMask(FieldMasks.allSetFieldsOf(adGroupCriterion))
+        .build();
+
+    // Creates the ad group criterion service.
+    try (AdGroupCriterionServiceClient adGroupCriterionServiceClient =
+        googleAdsClient.getLatestVersion().createAdGroupCriterionServiceClient()) {
+      // Updates the ad group criterion.
+      MutateAdGroupCriteriaResponse response = adGroupCriterionServiceClient.mutateAdGroupCriteria(
+          Long.toString(customerId),
+          ImmutableList.of(operation));
+      // Prints the results.
+      System.out.printf(
+          "Successfully updated the bid for ad group criterion with resource name '%s'.%n",
+          response.getResults(0).getResourceName());
+    }
+  }
+
+  /**
+   * Removes all ad group criteria targeting a user list under a given campaign. This is a
+   * necessary step before targeting a user list at the campaign level.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param campaignId the campaign under which to remove the ad group criteria.
+   */
+  private void removeExistingListCriteriaFromAdGroup(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      long campaignId) {
+    // Retrieves all of the ad group criteria under a campaign.
+    List<String> adGroupCriteria =
```

Comment:
Is this wrapping correct? Hard to see on Github. Would suggest `google-java-format -i`.